server.go

```diff
@@ -27,6 +27,10 @@ type Server struct {
 	AllowInsecureAuth bool
 	Debug             io.Writer
 
+	// If set, the AUTH command will not be advertised and authentication
+	// attempts will be rejected. This setting overrides AllowInsecureAuth.
+	AuthDisabled bool
```

Comment:
The main advantage of doing it like this, I feel, is that it mirrors the `AllowInsecureAuth` parameter. I don't feel too strongly about it though.

Reply:
Yeah, in the end you're right. Using an optional interface is silly.